### PR TITLE
Gabe

### PR DESCRIPTION
**Pull request by gabewilliam**
Created domain files for each of the DB tables (currency, user_detail, ledger, block and block_detail)
Most are pretty much identical to their DDL counterparts

### CODE REVIEW

File: src/main/scala/domain/Block.scala [hunk 1]
```diff
@@ -0,0 +1,8 @@
+package domain
+
+case class BlockId (id: Int)
+
+case class Block (id: BlockId,
+                  description: String,
+                  timeStamp: String,
+                  effectiveDate: String)
```
Comment (ben-mulhern) [inline src/main/scala/domain/Block.scala]: We discussed Block also containing a List of BlockDetail entries.

File: src/main/scala/domain/BlockDetail.scala [hunk 2]
```diff
@@ -0,0 +1,10 @@
+package domain
+
+case class BlockDetail(id: BlockId, //'Block.scala'
+                       entryNumber: Int,
+                       cashValue: Float,
```
Comment (ben-mulhern) [inline src/main/scala/domain/BlockDetail.scala]: I'd favour Double over Float.

File: src/main/scala/domain/Ledger.scala [hunk 3]
```diff
@@ -0,0 +1,6 @@
+package domain
+
+case class LedgerId(id: Option[String])
```
Comment (ben-mulhern) [inline src/main/scala/domain/Ledger.scala]: Should this be an option? When wouldn't we have a value?

File: src/main/scala/domain/User.scala [hunk 4]
```diff
@@ -0,0 +1,12 @@
+package domain
+
+case class UserId(id: String)
+
+case class Password(password: String)
+
+case class User( id: UserId,
+                 name: String,
+                 password: Password,
```
Comment (ben-mulhern) [inline src/main/scala/domain/User.scala]: I don't think we need the password and salt to exist in the User class - it's a security risk, we'd never want to send that data back to the browser.

**Comment by gabewilliam** [comment]
Removed password & salt from user
Cash value in Block is now double
Block contains list of BlockDetails
LedgerID no longer optional field

**Review by JohnHunt1990 — APPROVED** [review]

File: src/main/scala/domain/Block.scala [hunk 5]
```diff
@@ -0,0 +1,9 @@
+package domain
+
+case class BlockId (id: Int)
+
+case class Block (id: BlockId,
```
Comment (KarlFinn) [inline src/main/scala/domain/Block.scala]: Make id an option[BlockId] to cope with uncommitted blocks

File: src/main/scala/domain/BlockDetail.scala [hunk 6]
```diff
@@ -0,0 +1,10 @@
+package domain
+
+case class BlockDetail(id: BlockId, //'Block.scala'
```
Comment (KarlFinn) [inline src/main/scala/domain/BlockDetail.scala]: id is not needed in block detail as can be derived by container block

**Review by KarlFinn — CHANGES_REQUESTED** [review]

**Comment by gabewilliam** [comment]
Applied the changes Karl suggested

**Review by KarlFinn — APPROVED** [review]

**Pull request merged**